### PR TITLE
Ref Task: 6223 create home activity part 2

### PR DESCRIPTION
**Pull request by trieuthan92pro**
Link: https://edu-redmine.framgia.vn/issues/6223
screenshot_2019-02-13-10-09-41-739_com example soundcloud

### CODE REVIEW

**Comment by trieuthan92pro** [comment]
Ready

File: app/src/main/java/com/example/soundcloud/data/TabInfo.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.example.soundcloud.data;
+
+public class TabInfo {
+    private int mTabTextId;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/data/TabInfo.java]:
```suggestion
    private int mTextResource;
```

File: app/src/main/java/com/example/soundcloud/data/TabInfo.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private int mTabIconId;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/data/TabInfo.java]:
```suggestion
    private int mImageResource;
```

File: app/src/main/java/com/example/soundcloud/discover_screen/DiscoverFragment.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.example.soundcloud.discover_screen;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/discover_screen/DiscoverFragment.java]:
```suggestion
package com.example.soundcloud.discover;
```

File: app/src/main/java/com/example/soundcloud/home_screen/HomeActivity.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package com.example.soundcloud.home_screen;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/home_screen/HomeActivity.java]:
```suggestion
package com.example.soundcloud.home;
```

File: app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+import com.example.soundcloud.data.TabInfo;
+import com.example.soundcloud.discover_screen.DiscoverFragment;
+
+public class HomeFragmentPagerAdapter extends FragmentPagerAdapter {
+    private final int PAGE_COUNT = 3;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java]: hằng số a khai báo staic final 
```suggestion
    private static final int PAGE_COUNT = 3;
```

File: app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    private TabInfo[] mTabInfos;
+    private HomeScreenContract.Presenter mPresenter;
+    private Context context;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java]:
```suggestion
    private Context mContext;
```

File: app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public int getCount() {
+        return PAGE_COUNT;
+    }
+
+    @Nullable
+    @Override
+    public CharSequence getPageTitle(int position) {
+            TabInfo tabInfo = mTabInfos[position];
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java]: format code

File: app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    private final int PAGE_COUNT = 3;
+    private TabInfo[] mTabInfos;
+    private HomeScreenContract.Presenter mPresenter;
+    private Context context;
+
+    public HomeFragmentPagerAdapter(Context context, FragmentManager fm) {
+        super(fm);
+        this.context = context;
+        mPresenter = new HomeScreenPresenter();
+        mTabInfos = mPresenter.getTabsInfo();
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/home_screen/HomeFragmentPagerAdapter.java]: để giảm sự kết dính giữa các module
a nên inject tab info vào adapter thông qua constructor

File: app/src/main/java/com/example/soundcloud/splash_screen/SplashActivity.java [hunk 9]
```diff
@@ -39,4 +43,13 @@ private void initView() {
...
         mPresenter.start();
     }
+
+    private void goToHomeScreen() {
+        new Handler().postDelayed(() -> {
+            Intent intent = new Intent(this, HomeActivity.class);
+            startActivity(intent);
+            overridePendingTransition(R.anim.anim_in, R.anim.anim_out);
+            finish();
+        }, 3000);
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/soundcloud/splash_screen/SplashActivity.java]: https://www.bignerdranch.com/blog/splash-screens-the-right-way/

**Comment by trieuthan92pro** [comment]
Ready

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**